### PR TITLE
Use time.Duration in yaml config for intervals and fix unclosed respo…

### PR DESCRIPTION
**Pull request by boyang9527**
…nse body in test cases

[#130876133]

- change interval config to time.Duration
- replace  "github.com/cloudfoundry-incubator/candiedyaml" with  "gopkg.in/yaml.v2"
- fix unclosed response body in test cases

### CODE REVIEW

File: src/autoscaler/metricscollector/config/config_test.go [hunk 1]
```diff
@@ -39,29 +39,34 @@ server:
 			})
 
 			It("returns an error", func() {
-				Expect(err).To(BeAssignableToTypeOf(&candiedyaml.ParserError{}))
+				Expect(err).To(MatchError(MatchRegexp("yaml:*")))
```
Comment (fraenkel) [inline src/autoscaler/metricscollector/config/config_test.go]: This is not a correct regexp.

File: .gitmodules [hunk 2]
```diff
@@ -10,9 +10,6 @@
 [submodule "src/github.com/onsi/gomega"]
```
Comment (fraenkel) [inline .gitmodules]: You didn't submodule in goyaml.

File: src/autoscaler/metricscollector/config/config_test.go (hunk 1 above)
Comment (boyang9527) [inline src/autoscaler/metricscollector/config/config_test.go]: sorry my careless, correct it with .*

File: .gitmodules (hunk 2 above)
Comment (boyang9527) [inline .gitmodules]: the package was already there. It was added by aggregator/eventgen before.

**Pull request merged**